app/assets/javascripts/3dbio_viewer/src/webapp/components/RootViewerContents.tsx

```diff
@@ -77,6 +90,10 @@ export const RootViewerContents: React.FC<RootViewerContentsProps> = React.memo(
         }
     }, [pdbLoader.type, showLoading, hideLoading]);
 
+    React.useLayoutEffect(() => {
+        setInnerWidth(window.innerWidth);
+    }, []);
+
```

Comment:
All the new code added form a unity, right? I'd abstract it in a custom hook (`useResizableToolbar` or something like that).